### PR TITLE
Replace htmlunit with playwright

### PR DESCRIPTION
**Pull request by kshpak**
Playwright provides better API and has better JS support than htmlunit
Documentation is also better https://playwright.dev/java/docs/selectors/#quick-guide

Trello card with description of effort: https://trello.com/c/gqAsh7s6/1083-update-startstopts-tests-for-codequarkus-site-there-was-set-of-updates-to-enable-quarkiverse-extensions

### CODE REVIEW

**Comment by kshpak** [comment]
CI failures are not related to this PR. Check Daily CI fails with the same errors

**Comment by kshpak** [comment]
This library gives you an object of Locator anytime, but whenever you perform some action on Locator then respective ElementHandler is checked. So you must trigger some action on Locator to see if the element indeed exists

**Review by jsmrcka — CHANGES_REQUESTED** [review]

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java [hunk 1]
```diff
@@ -106,31 +114,24 @@ public void validateQuarkusVersionMatch(TestInfo testInfo) throws Exception{
...
-            throw new RuntimeException("Cannot load the page: " + webPageUrl);
-        }
-        JavaScriptJobManager manager = page.getEnclosingWindow().getJobManager();
-        while (manager.getJobCount() > 0 && System.currentTimeMillis() < endTime) {
+        Page page = browserContext.newPage();
+        page.setDefaultTimeout(timeoutSeconds * 1000);
+        page.navigate(url);
+        while (page.waitForSelector(pageLoadedSelector) == null) {
             Thread.sleep(1000);
         }
```
[the remark this answers is not in the document]
Comment (kshpak) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java:134]: you're right

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java [hunk 2]
```diff
@@ -32,63 +36,67 @@
...
 
     @Test
     public void validatePresenceOfIcon(TestInfo testInfo) throws Exception {
-        final HtmlPage page = loadPage(webPageUrl, 60);
+        Page page = loadPage(webPageUrl, 60);
         LOGGER.info("Trying to find element: " + elementIconByXpath);
-        HtmlElement shortcutIcon = page.getFirstByXPath(elementIconByXpath);
-        assertNotNull(shortcutIcon, "Element: " + elementIconByXpath + " is missing!");
+        Locator icon = page.locator(elementIconByXpath);
+        assertTrue(icon.count() == 1, "Element: " + elementIconByXpath + " is missing!");
```
[the remark this answers is not in the document]
Comment (kshpak) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java:74]: agree

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java [hunk 3]
```diff
@@ -32,63 +36,67 @@
  * Tests for checking presence of element on webpage
  */
 @Tag("codequarkus")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
[the remark this answers is not in the document]
Comment (kshpak) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java:39]: BrowserContext is not so expensive, but I think we don't need to create a new instance of browser each time. Also IMO it's better to leave `loadPage()` inside test methods, that way we create a new chrome tab for every test

[the remark this answers is not in the document]
Comment (jsmrcka) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java:36]: You're right. Now that I look at it, it's better to have fresly loaded page every time we locate an element or wait for its state.

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java [hunk 4]
```diff
@@ -106,31 +111,22 @@ public void validateQuarkusVersionMatch(TestInfo testInfo) throws Exception{
...
-        } catch (Exception e) {
-            throw new RuntimeException("Cannot load the page: " + webPageUrl);
-        }
-        JavaScriptJobManager manager = page.getEnclosingWindow().getJobManager();
-        while (manager.getJobCount() > 0 && System.currentTimeMillis() < endTime) {
-            Thread.sleep(1000);
-        }
-
+        Page page = browserContext.newPage(); // this will create new tab inside browser
+        page.setDefaultTimeout(10 * 1000);
```
Comment (jsmrcka) [inline testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java:127]:
```suggestion
        page.setDefaultTimeout(timeoutSeconds * 1000);
```
The original timeout was `timeoutSeconds * 1000`, with 1000ms sleeps.

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java [hunk 5]
```diff
@@ -32,67 +33,71 @@
...
 
     @Test
-    public void validatePresenceOfTitle(TestInfo testInfo) throws Exception {
-        final HtmlPage page = loadPage(webPageUrl, 60);
+    public void validatePresenceOfTitle(TestInfo testInfo) {
+        Page page = loadPage(webPageUrl, 60);
         LOGGER.info("Verify page title is: " + elementTitleByText);
-        assertEquals(page.getTitleText(), elementTitleByText,
-                "Title doesn't match. Found on the web: " + page.getTitleText() + ". Expected: " + elementTitleByText);
+        assertEquals(page.title(), elementTitleByText,
```
Comment (jsmrcka) [inline testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java:78]:
```suggestion
        assertEquals(elementTitleByText, page.title(),
```
`Expected` and `actual` are swapped.

**Review by jsmrcka — CHANGES_REQUESTED** [review]

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java (hunk 4 above)
Comment (kshpak) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java]: good one

**Review by jsmrcka — APPROVED** [review]

**Pull request merged**